plugins/ltac/pptactic.ml

```diff
@@ -492,7 +492,18 @@ let string_of_genarg_arg (ArgumentType arg) =
         (prlist_with_sep (fun () -> str",")
            (fun id -> spc () ++ pr_hyp_location pr_id id) l ++ pr_occs)
 
-  let pr_orient b = if b then mt () else str "<- "
+
+  (* When the [ssreflect.SsrSynax] module is imported, ssreflect operates
+     in reduced compatibility mode. During printing, we try to account for
+     this when this module is imported. See [plugins/ssr/ssrparser.mlg] for
+     the code that enables the reduced compatibility mode. *)
+  let ssr_loaded = Summary.ref ~name:"pptactic-ssr-loaded" false
+  let is_ssr_loaded () =
```

Comment:
This kind of hacks made sense when ssr was not part of the Coq codebase, but we can do better now. I think that instead you should perform a control inversion by moving this as a hook in Ltac, and register the hook from ssr. This would allow sharing the same horrible code in ssr in a single place.

Reply:
Good idea. Done.